tutorials/gui/custom_gui_controls.rst

```diff
@@ -100,14 +131,39 @@ This function is
 :ref:`Control._gui_input() <class_Control__gui_input>`.
 Simply override it in your control. No processing needs to be set.
 
-::
+.. tabs::
+ .. code-tab:: gdscript GDScript
 
     extends Control
 
     func _gui_input(event):
        if event is InputEventMouseButton and event.button_index == BUTTON_LEFT and event.pressed:
            print("Left mouse button was pressed!")
 
+ .. code-tab:: csharp
+
+    public override void _GuiInput(InputEvent @event)
+    {
+        var mouseButtonEvent = @event as InputEventMouseButton;
+        if (mouseButtonEvent != null)
+        {
+            if (mouseButtonEvent.ButtonIndex == (int)ButtonList.Left && mouseButtonEvent.Pressed)
+            {
+                GD.Print("Left mouse button was pressed!");
+            }
+        }
+    }
+
+    // or if using c# 7 or greater
+    public override void _GuiInput(InputEvent @event)
+    {
+        if (@event is InputEventMouseButton mbe && mbe.ButtonIndex == (int)ButtonList.Left && mbe.Pressed)
```

Comment:
Is it alright to use C# 7 feature (pattern matching, in this case) in samples? I remember you recommended stay with C# 6 in engine sources (https://github.com/godotengine/godot-docs/issues/1296#issuecomment-379631679) but if it's ok with samples, I guess we can also follow the same pattern in L150.

Reply:
Here there are two implementations L145-155 are code that should be compatible with all c# versions.

Then L157-165 show how it could be approached if pattern matching is available.

If this is unclear I will be more explicit in the description .

Reply:
No, it was my fault that I totally missed that comment. Sorry, shouldn't have done any code reading right after I wake up in the morning :blush: